### ISSUE TITLE
2.5.0 build failure

### ISSUE THREAD

**Issue by chenrui333** [dreibh/subnetcalc#16]
👋 trying to build the latest release, but run into some build issue. The error log is as below:

<details>
<summary>error build log</summary>

```
  /tmp/subnetcalc-20240210-3255-d4xokh/subnetcalc-subnetcalc-2.5.0/src/tools.cc:132:36: error: no member named 's6_addr16' in 'in6_addr'
     return ntohs(address->sin6_addr.s6_addr16[0]) == 0x64 &&
                  ~~~~~~~~~~~~~~~~~~ ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/sys/_endian.h:132:46: note: expanded from macro 'ntohs'
  #define ntohs(x)        __DARWIN_OSSwapInt16(x)
                                               ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/libkern/_OSByteOrder.h:85:40: note: expanded from macro '__DARWIN_OSSwapInt16'
      ((__uint16_t)(__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt16(x) : _OSSwapInt16(x)))
                                         ^
  /tmp/subnetcalc-20240210-3255-d4xokh/subnetcalc-subnetcalc-2.5.0/src/tools.cc:132:36: error: no member named 's6_addr16' in 'in6_addr'
     return ntohs(address->sin6_addr.s6_addr16[0]) == 0x64 &&
                  ~~~~~~~~~~~~~~~~~~ ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/sys/_endian.h:132:46: note: expanded from macro 'ntohs'
  #define ntohs(x)        __DARWIN_OSSwapInt16(x)
                                               ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/libkern/_OSByteOrder.h:85:71: note: expanded from macro '__DARWIN_OSSwapInt16'
      ((__uint16_t)(__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt16(x) : _OSSwapInt16(x)))
                                                                        ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/libkern/_OSByteOrder.h:44:34: note: expanded from macro '__DARWIN_OSSwapConstInt16'
      ((__uint16_t)((((__uint16_t)(x) & 0xff00U) >> 8) | \
                                   ^
```

</details>

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/7855958025/job/21438280388?pr=162285
relates to Homebrew/homebrew-core#162285

**Comment by dreibh** [comment]
Confirmed. I can reproduce the issue under FreeBSD. You can already check branch "dreibh/build-fix-12feb2024" with a fix.

**Comment by dreibh** [comment]
The problem should be fixed in version 2.5.1.
